mpi-proxy-split/test/Makefile

```diff
@@ -147,10 +147,11 @@ check-integrated_dmtcp_text: tidy integrated_dmtcp_test
 	 $(MPICC) -g3 -O0 -c -o $@ $< $(MPI_CFLAGS)
 
 %.exe: %.f90
```

Comment:
Could you please add a small note here stating why are we using -fallow-argument-mismatch?
Also, please add that this flag is not affecting the execution with GNU and CRAY MPICH.